### ISSUE TITLE
Nested \tfrac repeatedly increases font-size.

### ISSUE THREAD

**Issue by mbostock** [Khan/KaTeX#1352]
In v0.10.0-alpha, it appears that nested \tfrac causes the font-size to balloon. For example:

```tex
\tfrac{1}{1+\tfrac{1}{1+\tfrac{1}{1+\tfrac{1}{1+\tfrac{1}{1+\tfrac{1}{1+\tfrac{1}{1+\tfrac{1}{2}}}}}}}}
```

![image](https://user-images.githubusercontent.com/230541/40497244-99e0fa2a-5f30-11e8-9be3-234c8d7a18f5.png)

Switching to \dfrac or \frac works around the issue:

![image](https://user-images.githubusercontent.com/230541/40497337-ca10e1b0-5f30-11e8-936a-a6a9e65a9be8.png)

![image](https://user-images.githubusercontent.com/230541/40497311-c0467dfc-5f30-11e8-94ba-6af2c34264bb.png)

Looking at the applied style, I think the issue is that it’s using em’s for font-size, which means that it’s relative to the containing font-size, and thus applies multiplicatively when nested rather than truly resetting the font-size.

```css
.katex .fontsize-ensurer.reset-size3.size6,
.katex .sizing.reset-size3.size6 {
    font-size: 1.42857143em;
}
```

Perhaps [rems](https://developer.mozilla.org/en-US/docs/Web/CSS/length#rem) would be appropriate here, rather than ems?

**Comment by ronkok** [comment]
It also occurs in v0.9.0.

**Comment by ronkok** [comment]
But nested `\frac` works properly.

**Comment by ronkok** [comment]
We're repeatedly applying a size multiplier of 1/0.7 when we should only do that once. Behavior becomes better in this example if in `genfrac.js` we change line 78 from

```
        } else if (group.value.size === "text") {
```

to 

```
       } else if (group.value.size === "text" && style.size < Style.TEXT.size ) {
```

I think that also addresses the other possible combinations of style and size. But I would want to think harder about that before committing.

**Comment by ronkok** [comment]
Fixed by #1378, so closing the issue.
